### PR TITLE
ESWE-169: detailed attendance data - QA fix

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/prisonapi/OffenderAttendance.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.justice.digital.hmpps.whereabouts.dto.prisonapi
+
+/**
+ * Data from prison api
+ */
+data class OffenderAttendance(
+  var eventDate: String,
+  var outcome: String? = null,
+  val prisonId: String? = null,
+  val activity: String? = null,
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/prisonapi/OffenderAttendance.kt:10]: Turns out activity can be missing in the data from prison-api

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.kt [hunk 2]
```diff
@@ -55,18 +54,14 @@ class AttendanceService(
     val attendance = attendanceRepository
       .findByPrisonIdAndEventLocationIdAndEventDateAndPeriod(prisonId, eventLocationId, date, period)
 
-    return attendance
-      .map { toAttendanceDto(it) }
-      .toSet()
+    return attendance.map(this::toAttendanceDto).toSet()
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.kt:57]: Some tidying

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/PrisonApi.java [hunk 3]
```diff
@@ -83,14 +83,14 @@ public void putAttendanceForMultipleBookings(final Set<BookingActivity> bookingA
     }
 
     public Page<OffenderAttendance> getAttendanceForOffender(final String offenderNo, final LocalDate fromDate, final LocalDate toDate,
-                                                             final Optional<String> outcome, final Pageable pageable) {
+                                                             final String outcome, final Pageable pageable) {
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/PrisonApi.java:86]: Bad style to have Optional as a param apparently

Comment (lbennett-moj) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/PrisonApi.java:86]: i was confused by a java file amongst the kotlin!

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**